### ISSUE TITLE
Coadding failure for some exposure from 20200219

### ISSUE THREAD

**Issue by rongpu** [desihub/desispec#893]
@sbailey When running the following commands to create a 2-exposure coadd

```
source /global/cfs/cdirs/desi/software/desi_environment.sh master
export REDUXDIR=/global/cfs/cdirs/desi/spectro/redux/daily
export OUTDIR=$SCRATCH/desi/minisv2/2_exp_coadd
mkdir -p $OUTDIR
time desi_coadd_spectra --coadd-cameras -i $REDUXDIR/exposures/20200219/00051059/cframe-[brz]0-*.fits $REDUXDIR/exposures/20200219/00051060/cframe-[brz]0-*.fits  -o $OUTDIR/70003/coadd-0-2exp-subset-3.fits
```
I get the following error message:

```
INFO:coadd_spectra.py:51:main: reading spectra ...
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/code/desispec/master/bin/desi_coadd_spectra", line 16, in <module>
    coadd_spectra.main(args)
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/code/desispec/master/py/desispec/scripts/coadd_spectra.py", line 63, in main
    spectra = frames2spectra(frames)
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/code/desispec/master/py/desispec/pixgroup.py", line 507, in frames2spectra
    fibermap = np.hstack(fibermap)
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/conda/lib/python3.6/site-packages/numpy/core/shape_base.py", line 338, in hstack
    return _nx.concatenate(arrs, 0)
TypeError: invalid type promotion
```
This only happens for some of the exposures. E.g. this command runs successfully:
```
time desi_coadd_spectra --coadd-cameras -i $REDUXDIR/exposures/20200219/00051053/cframe-[brz]0-*.fits $REDUXDIR/exposures/20200219/00051054/cframe-[brz]0-*.fits  -o $OUTDIR/70003/coadd-0-2exp-subset-0.fits
```

**Comment by sbailey** [comment]
Thanks for the report, @rongpu .  This is caused by the fibermap format mismatch described in issue #887.

As a temporary workaround I pushed a branch "coadd-fibermaps" that does the necessary reformatting while doing the coadd, but the better long term solution (i.e. next week) is to fix the format of the fibermaps themselves and rerun.

Please try using the coadd-fibermaps branch to make your coadds.  ping @akremin if you need help configuring your own version of desispec at NERSC.

**Comment by sbailey** [comment]
fixed by @akremin in PR #902
